inspire_mitmproxy/http/response.py

```diff
@@ -0,0 +1,108 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE-MITMPROXY.
+# Copyright (C) 2018 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+"""INSPIRE-MITMProxy internal HTTP request representation."""
+
+from typing import Any, Dict, Optional
+
+from mitmproxy.http import HTTPResponse
+from mitmproxy.net.http.status_codes import RESPONSES
+
+from .headers import MITMHeaders
+from .utils import encoding_by_header
+
+
+class MITMResponse:
+    def __init__(
+        self,
+        status_code: int = 200,
+        status_message: Optional[str] = None,
+        body: Optional[str] = None,
+        headers: MITMHeaders = None,
+        original_encoding: Optional[str] = None,
+        http_version: Optional[str] = None,
+    ) -> None:
+        self.status_code = status_code
+        self.status_message = status_message or RESPONSES[status_code]
+        self.body = body or ''
+        self.headers = headers or MITMHeaders({})
+        self.original_encoding = original_encoding
+        self.http_version = http_version or 'HTTP/1.1'
```

Comment:
Guess the encoding from the headers here if not passed, and not in the 'from_*' methods:
```
 +        encoding = encoding_by_header(MITMHeaders.from_mitmproxy(response.headers)) 
~``